include/mbgl/storage/response.hpp

```diff
@@ -35,7 +35,7 @@ class Response {
     optional<std::string> etag;
 
     bool isFresh() const {
-        return !expires || *expires > util::now();
+        return (!expires || *expires > util::now()) && !error;
```

Comment:
Should this condition be `expires ? *expires > util::now() : !error`? In case there is an explicit `Cache-Control` or `Expires` header on an error response, it [should be respected](http://stackoverflow.com/questions/40006148/can-a-500-internat-server-error-response-be-cached-by-a-browser).

Also, `error` can be `Reason::NotFound` (404). We should carefully consider (and test) what the retry behavior is in that situation.